src/status_im/events.cljs

```diff
@@ -115,30 +115,39 @@
 (fx/defn system-theme-mode-changed
   {:events [:system-theme-mode-changed]}
   [{:keys [db] :as cofx} theme]
-  (let [cur-theme     (get-in db [:multiaccount :appearance])
-        current-tab   (get db :current-tab :chat)
-        view-id       (:view-id db)
-        screen-params (get-in db [:navigation/screen-params view-id])
-        root-id       @navigation.core/root-id]
+  (let [cur-theme      (get-in db [:multiaccount :appearance])
+        current-tab    (get db :current-tab :chat)
+        view-id        (:view-id db)
+        screen-params  (get-in db [:navigation/screen-params view-id])
+        root-id        @navigation.core/root-id
+        dispatch-later (cond-> []
+                         (= view-id :chat)
+                         (conj {:ms       1000
+                                :dispatch [:chat.ui/navigate-to-chat (:current-chat-id db)]})
+
+                         (and
+                          (= root-id :chat-stack)
+                          (not-any? #(= view-id %) '(:home :empty-tab :wallet :status :my-profile :chat)))
+                         (conj {:ms       1000
+                                :dispatch [:navigate-to view-id screen-params]})
+
+                         (some #(= view-id %) navigation.core/community-screens)
+                         (conj {:ms 800 :dispatch
+                                [:navigate-to :community
+                                 (get-in db [:navigation/screen-params :community])]})
```

Comment:
When the system theme changes, we are restoring the app state. If earlier view-id is related to any community-related screen, we are making sure to first open community before opening that screen. The above review screens are edge cases, (the user was editing the community when the system theme changed), but we can include these two also to cover those edge cases? (change seems too small to do in separate pr). wdyt?